src/mame/video/segasaturn_vdp1.h

```diff
@@ -0,0 +1,286 @@
+   ---- ---- ---- -x-- | Double Interlace Draw Line (DIL)
+   ---- ---- ---- --x- | Frame Buffer Change Trigger (FCM)
+   ---- ---- ---- ---x | Frame Buffer Change Mode (FCT)
+*/
+#define STV_VDP1_FBCR ((m_vdp1_regs[0x002/2] >> 0)&0xffff)
+#define STV_VDP1_EOS ((STV_VDP1_FBCR & 0x0010) >> 4)
+#define STV_VDP1_DIE ((STV_VDP1_FBCR & 0x0008) >> 3)
+#define STV_VDP1_DIL ((STV_VDP1_FBCR & 0x0004) >> 2)
+#define STV_VDP1_FCM ((STV_VDP1_FBCR & 0x0002) >> 1)
+#define STV_VDP1_FCT ((STV_VDP1_FBCR & 0x0001) >> 0)
+
+/*Plot Trigger Register*/
+/*
+   xxxx xxxx xxxx xx-- | UNUSED
+   ---- ---- ---- --xx | Plot Trigger Mode (PTM)
+
+   Plot Trigger Mode:
+   3 Invalid
+   2 Automatic draw
+   1 VDP1 draw by request
+   0 VDP1 Idle (no access)
+*/
+#define STV_VDP1_PTMR ((m_vdp1_regs[0x004/2])&0xffff)
+#define STV_VDP1_PTM  ((STV_VDP1_PTMR & 0x0003) >> 0)
+#define PTM_0         m_vdp1_regs[0x004/2]&=~0x0001
+
+/*
+    Erase/Write Data Register
+    16 bpp = data
+    8 bpp = erase/write data for even/odd X coordinates
+*/
+#define STV_VDP1_EWDR ((m_vdp1_regs[0x006/2])&0xffff)
+
+/*Erase/Write Upper-Left register*/
+/*
+   x--- ---- ---- ---- | UNUSED
+   -xxx xxx- ---- ---- | X1 register
+   ---- ---x xxxx xxxx | Y1 register
+
+*/
+#define STV_VDP1_EWLR ((m_vdp1_regs[0x008/2])&0xffff)
+#define STV_VDP1_EWLR_X1 ((STV_VDP1_EWLR & 0x7e00) >> 9)
+#define STV_VDP1_EWLR_Y1 ((STV_VDP1_EWLR & 0x01ff) >> 0)
+/*Erase/Write Lower-Right register*/
+/*
+   xxxx xxx- ---- ---- | X3 register
+   ---- ---x xxxx xxxx | Y3 register
+
+*/
+#define STV_VDP1_EWRR ((m_vdp1_regs[0x00a/2])&0xffff)
+#define STV_VDP1_EWRR_X3 ((STV_VDP1_EWRR & 0xfe00) >> 9)
+#define STV_VDP1_EWRR_Y3 ((STV_VDP1_EWRR & 0x01ff) >> 0)
+/*Transfer End Status Register*/
+/*
+   xxxx xxxx xxxx xx-- | UNUSED
+   ---- ---- ---- --x- | CEF
+   ---- ---- ---- ---x | BEF
+
+*/
+#define STV_VDP1_EDSR ((m_vdp1_regs[0x010/2])&0xffff)
+#define STV_VDP1_CEF  (STV_VDP1_EDSR & 2)
+#define STV_VDP1_BEF  (STV_VDP1_EDSR & 1)
+/**/
+#define CEF_1   m_vdp1_regs[0x010/2]|=0x0002
+#define CEF_0   m_vdp1_regs[0x010/2]&=~0x0002
+#define BEF_1   m_vdp1_regs[0x010/2]|=0x0001
+#define BEF_0   m_vdp1_regs[0x010/2]&=~0x0001
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+class saturn_vdp1_device : public device_t
+{
+public:
+	// construction/destruction
+	saturn_vdp1_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// I/O operations
+	void regs_w(offs_t offset, u16 data, u16 mem_mask = ~0);
+	u16 regs_r(offs_t offset);
+	void vram_w(offs_t offset, u32 data, u32 mem_mask = ~0);
+	u32 vram_r(offs_t offset);
+	u32 framebuffer0_r(offs_t offset, u32 mem_mask = ~0);
+	void framebuffer0_w(offs_t offset, u32 data, u32 mem_mask = ~0);
+	template<class T> void set_hostcpu(T &&tag) { m_host_cpu.set_tag(std::forward<T>(tag)); }
```

Comment:
Configuration member functions shouldn’t be mixed up amongst live I/O functions –  most people prefer to put them below the constructor.